src/Analyser/Scope.php

```diff
@@ -634,9 +634,9 @@ function (Expr\ArrayItem $item): Type {
 		}
 
 		if ($node instanceof Expr\ArrayDimFetch && $node->dim !== null) {
-			$arrayType = $this->getType($node->var);
-			if ($arrayType instanceof ArrayType) {
-				return $arrayType->getItemType();
+			$offsetAccessibleType = $this->getType($node->var);
+			if ($offsetAccessibleType->isOffsetAccesible()->yes()) {
+				return $offsetAccessibleType->getOffsetValueType();
```

Comment:
Ideally it should be possible to drop the condition and simplify the code to

~~~php
$this->getType($node->var)->getOffsetValueType();
~~~

Reply:
If I remove condition, result of `Traversable&int[]` is `int`. Maybe it is correct and we need improve some rule which will check similar not-safe cases. What do you think?